File: 60. Permutation Sequence/60. Permutation Sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> factorial(10, 1);
        vector<int> nums(n);
        string result;
        
        for (int i = 1; i <= n; i++) {
            factorial[i] = factorial[i-1] * i;
            nums[i-1] = i;
        }
        
        k--;  // Convert to 0-based index
        
        for (int i = 0; i < n; i++) {
            int index = k / factorial[n-1-i];
            result += to_string(nums[index]);
            nums.erase(nums.begin() + index);
            k %= factorial[n-1-i];
        }
        
        return result;
    }
};

File: 60. Permutation Sequence/60. Permutation Sequence.rs
impl Solution {
    pub fn get_permutation(n: i32, mut k: i32) -> String {
        let mut factorial = vec![1; 10];
        let mut nums: Vec<i32> = (1..=n).collect();
        let mut result = String::new();
        
        for i in 1..=n as usize {
            factorial[i] = factorial[i-1] * i as i32;
        }
        
        k -= 1;  // Convert to 0-based index
        
        for i in 0..n as usize {
            let index = (k / factorial[n as usize - 1 - i]) as usize;
            result.push_str(&nums[index].to_string());
            nums.remove(index);
            k %= factorial[n as usize - 1 - i];
        }
        
        result
    }
}